### ISSUE TITLE
[Bug] [0.35.0~0.36.0] The right-click menu in Apple device displays abnormally (no background) 

### ISSUE THREAD

**Issue by SuperManito** [microsoft/monaco-editor#3602]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [X] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

**Comment by SuperManito** [comment]
Some style variables are not defined

**Comment by hediet** [comment]
Please fill out the bug report form completely and provide screenshots. Thanks!

**Comment by SuperManito** [comment]
> Please fill out the bug report form completely and provide screenshots. Thanks!

![image](https://user-images.githubusercontent.com/68613938/221744071-b79939ce-77f1-4a09-a001-08fdaa5ca6be.png)

**Comment by hediet** [comment]
Unfortunately, I cannot reproduce. Can you look into what causes this?

**Comment by SuperManito** [comment]
> ~Unfortunately, I cannot reproduce. Can you look into what causes this?~
> 
> I can reproduce on the iPad.

Sorry. I can't help you analyze the code. I just report a bug. Maybe you can open this issue to get more help from community developers.

**Comment by catfan** [comment]
@haeguri Any update for this issue? This bug still exists on 0.39 and 0.40-dev.

This happened since 0.35, and 0.34.1 is fine.

We are looking forward to fixing it so that it can work on iOS, thanks!

**Comment by catfan** [comment]
Here is the screenshot of iOS Safari developer tools.
<img width="1369" alt="image" src="https://github.com/microsoft/monaco-editor/assets/1467904/7031df4e-41a7-43fd-83f2-3c4b88f6d9f9">

And here is the screenshot of desktop Chrome developer tools.
<img width="1598" alt="image" src="https://github.com/microsoft/monaco-editor/assets/1467904/72adace4-daea-484d-8b32-15cc1b5d3b7f">

We can see that all CSS variables prefixed with `--vscode` are not applied on iOS. That causes the background color `var(--vscode-menu-background)` of the menu container on iOS to be transparent `rgba(0,0,0,0)`.

**Comment by SuperManito** [comment]
> Here is the screenshot of iOS Safari developer tools. <img alt="image" width="1369" src="https://user-images.githubusercontent.com/1467904/252852855-7031df4e-41a7-43fd-83f2-3c4b88f6d9f9.png">
> 
> And here is the screenshot of desktop Chrome developer tools. <img alt="image" width="1598" src="https://user-images.githubusercontent.com/1467904/252853938-72adace4-daea-484d-8b32-15cc1b5d3b7f.png">
> 
> We can see that all CSS variables prefixed with `--vscode` are not applied on iOS. That causes the background color `var(--vscode-menu-background)` of the menu container on iOS to be transparent `rgba(0,0,0,0)`.

Yes, I found that CSS variables have become undefined. I did not find any relevant codes in this repo.

**Comment by Nyaasu66** [comment]
> > Here is the screenshot of iOS Safari developer tools. <img alt="image" width="1369" src="https://user-images.githubusercontent.com/1467904/252852855-7031df4e-41a7-43fd-83f2-3c4b88f6d9f9.png">
> > And here is the screenshot of desktop Chrome developer tools. <img alt="image" width="1598" src="https://user-images.githubusercontent.com/1467904/252853938-72adace4-daea-484d-8b32-15cc1b5d3b7f.png">
> > We can see that all CSS variables prefixed with `--vscode` are not applied on iOS. That causes the background color `var(--vscode-menu-background)` of the menu container on iOS to be transparent `rgba(0,0,0,0)`.
> 
> Yes, I found that CSS variables have become undefined. I did not find any relevant codes in this repo.

The relevant codes are here:
https://github.com/microsoft/vscode/blob/35573c42a59946d4ea0c5b533055f86a5585abbf/src/vs/platform/theme/common/colorRegistry.ts#L488

**Comment by ExtAnimal** [comment]
It reproduces in your own examples. Get it fixed.

![Screenshot 2024-04-11 at 07 24 02](https://github.com/microsoft/monaco-editor/assets/200516/e60b1d7d-f862-45a2-ab63-39444d2d36cb)

**Comment by SuperManito** [comment]
> It reproduces in your own examples. Get it fixed.
> 
> ![Screenshot 2024-04-11 at 07 24 02](https://private-user-images.githubusercontent.com/200516/321512603-e60b1d7d-f862-45a2-ab63-39444d2d36cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4MTQwMTQsIm5iZiI6MTcxMjgxMzcxNCwicGF0aCI6Ii8yMDA1MTYvMzIxNTEyNjAzLWU2MGIxZDdkLWY4NjItNDVhMi1hYjYzLTM5NDQ0ZDJkMzZjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMVQwNTM1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNGI5N2IyNDNhYzYwMTc1YzQ3OGM4MWMzMmZlOGFlZTg4Yzk4MzE1MWU0MzM1ZjhkMmUwY2FjNTY3MzhlNDQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6Rnn2uX8mN5ybNJasolnckpCHSF0jlBnA5d04og5TKM)

At present, official seems to be quite passive in addressing the issues occurring on mobile devices.

**Comment by ExtAnimal** [comment]
> Please fill out the bug report form completely and provide screenshots. Thanks!

Lazy. Just go to https://microsoft.github.io/monaco-editor/playground.html?source=v0.45.0#example-extending-language-services-configure-javascript-defaults on your iPad, or with Chrome emulating an iPad:

![Screenshot 2024-04-11 at 07 24 02](https://github.com/microsoft/monaco-editor/assets/200516/0310bcd8-d5bb-461c-a24e-2b8a66f39347)

**Comment by ExtAnimal** [comment]
> At present, official seems to be quite passive in addressing the issues occurring on mobile devices.

I added a quickfix to our editor component which we use to wrap a Monaco instance:

```
            // Workaround for https://github.com/microsoft/monaco-editor/issues/3602
            // Hoist the menu background var up one level because the context menu is placed outside of the editor.
            this.element.style.setProperty('--vscode-menu-background', getComputedStyle(this.element.querySelector('.monaco-editor')).getPropertyValue('--vscode-menu-background'));
```

**Comment by SuperManito** [comment]
> > 目前，官方在解决移动设备上发生的问题方面似乎相当被动。
> 
> 我为我们的编辑器组件添加了一个快速修复程序，我们用它来包装一个摩纳哥实例：
> 
> ```
> // Workaround for https://github.com/microsoft/monaco-editor/issues/3602
> // Hoist the menu background var up one level because the context menu is placed outside of the editor.
> this.element.style.setProperty('--vscode-menu-background', getComputedStyle(this.element.querySelector('.monaco-editor')).getPropertyValue('--vscode-menu-background'));
> ```

My project has scenarios for mobile use. Do you know why the monaco editor cannot long press the screen on the mobile end to summon system copy pop ups? In addition to this issue, it is also impossible to achieve cursor movement by long pressing the system keyboard space. I don't understand why the official doesn't optimize the experience of such a powerful editor on mobile devices

**Comment by catfan** [comment]
@hediet 

This issue has been fixed since 0.48.0-dev-20240329. However, the 0.48.0-rc has the problem again, but 0.49.0-dev-20240423 is working.

Is it about the building problem?

**Comment by munrocket** [comment]
I am fixed it here: #4914, background color for `.monaco-action-bar` is missing.

```
const MonacoWrapper = styled(Box)(({ theme }) => ({
    '.monaco-action-bar': {
        backgroundColor: `${theme.palette.background.paper} !important`
    }
}));

<MonacoWrapper>
  <Monaco editorOptions={ useShadowDOM: false } />
</ MonacoWrapper>
```

Someone need to make PR with fix, turning off shadowDOM not the best experience.

**Comment by SuperManito** [comment]
> I am fixed it here: [#4914](https://github.com/microsoft/monaco-editor/issues/4914), background color for is missing.`.monaco-action-bar`
> 
> ```
> const MonacoWrapper = styled(Box)(({ theme }) => ({
>     '.monaco-action-bar': {
>         backgroundColor: `${theme.palette.background.paper} !important`
>     }
> }));
> 
> <MonacoWrapper>
>   <Monaco editorOptions={ useShadowDOM: false } />
> </ MonacoWrapper>
> ```
> 
> Someone need to make PR with fix, turning off shadowDOM not the best experience.

Monaco Editor has not been updated for a long time. Maybe they are all busy developing Copilot extension.

**Comment by munrocket** [comment]
This bug has ticket `help wanted`, so if you will fix it with PR it will be merged.
